src/Main/Character/CharacterParses.js

```diff
@@ -0,0 +1,482 @@
+    }
+    return fetchWcl(`parses/character/${charName}/${charRealm}/${this.props.region}`, {
+      metric: this.state.metric,
+      zone: this.state.activeZoneID,
+      _: refresh ? +new Date() : undefined,
+    }).then((rawParses) => {
+      if (rawParses.status === 400) {
+        // means char was not found on WCL
+        this.setState({
+          error: true,
+        });
+        return;
+      }
+      if (this.state.class !== "") { //only update parses when class was already parsed (since its only a metric/raid change)
+        const parses = this.changeParseStructure(rawParses);
+        this.setState({
+          parses: parses,
+          isLoading: false,
+        });
+
+        return;
+      }
+
+      if (rawParses.length === 0) { //happens when the character has no logs for the selected raid
+        this.setState({
+          parses: [],
+          isLoading: false,
+        });
+        return;
+      }
+      const charClass = rawParses[0].specs[0].class;
+      const specs = Object.values(SPECS)
+        .map((elem, index) => { if (elem.className.replace(" ", "") !== charClass) { return undefined; } return elem.specName; })
+        .filter((elem, index) => { return elem; });
+
+      const parses = this.changeParseStructure(rawParses);
+      this.setState({
+        specs: [...new Set(specs)],
+        activeSpec: [...new Set(specs.map(elem => { return elem.replace(" ", ""); }))],
+        activeDifficulty: DIFFICULTIES.filter(elem => { return elem; }),
+        class: charClass,
+        parses: parses,
+        isLoading: false,
+      });
+    });
+  }
+
+  render() {
+    return (
+      <div className="container charparse">
+        <div className="flex-main">
+          <div className="row">
+            <div className="col-md-5">
+              <div className="panel">
+                <div className="flex-main">
+                  <div className="row filter">
+                    <div className="col-md-12" style={{ marginBottom: 20, position: 'relative', height: 280 }}>
+                      {this.state.image && (
+                        <div className="char-image">
+                          <img 
+                            src={`https://render-${this.props.region}.worldofwarcraft.com/character/${this.state.image}-main.jpg`}
+                            alt={"Pic"} 
+                            onError={e => this.setState({ image: false })}
+                            style={{ width: '100%' }} />
+                        </div>
+                      )}
+                      <h2 style={{ fontSize: '1.8em' }}>{this.props.region} - {this.props.realm}</h2>
+                      <h2 style={{ fontSize: '2.4em', margin: '10px 10px' }}>
+                        {this.props.name}
+                      </h2>
+                      {this.state.class && (
+                          <img
+                            src={`/specs/${this.state.class}-New.png`}
+                            alt="Pic"
```

Comment:
Please use a valid name, e.g. this.state.class

Reply:
fixed